eval/unwrap.go

```diff
@@ -97,12 +97,37 @@ func (u ValueUnwrapper) NonNegativeInt() (int, error) {
 	return i, u.err
 }
 
+func (u ValueUnwrapper) Fd() (int, error) {
+	s, err := u.String()
+	if err != nil {
+		return 0, err
+	}
+	switch s {
+	case "stdin":
+		return 0, nil
+	case "stdout":
+		return 1, nil
+	case "stderr":
+		return 2, nil
+	default:
+		i, errInt := u.NonNegativeInt()
```

Comment:
You can reuse the name `err` here. It is common pattern in Go unless you have to store multiple errors.